### ISSUE TITLE
Please consider having special error codes

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#3381]
I use both MyPy and Pyright on the same project.  In some rare instances, Pyright has a false positive where MyPy does not.  I mark these with `# type: ignore[pyright]`.

The problem is that when I call MyPy with `warn_unused_ignores = true`, I get things like:
```
cmm/visualization/main_window.py:42: error: Unused "type: ignore" comment
                application.aboutToQuit.connect(self.plot_thread.terminate)  # type: ignore[pyright]
```

In my ideal world, Pyright would have its own error codes, and MyPy would know to ignore those when warning.

See https://github.com/python/mypy/issues/12671 for the parallel MyPy issue, which is probably more important.

**Comment by erictraut** [comment]
Pyright emits over 600 different error messages, and we're continually adding and tweaking these. Adding a separate error code for each one would be a big maintenance headache, and we don't have any current plans to do this.

Pyright groups each of these error messages under about 60 different named "diagnostic rules". Each diagnostic is reported with its diagnostic rule.

I recently added a mechanism that I think will address your needs. You can now add a `# pyright: ignore` comment at the end of a line. You can also optionally add the name of one or more diagnostic rules in brackets, such as `# pyright: ignore[reportInvalidTypeVarUse]` to suppress that error. See [this documentation](https://github.com/microsoft/pyright/blob/main/docs/comments.md#line-level-diagnostic-suppression) for details. This mechanism also works with the `reportUnnecessaryTypeIgnoreComment` option, which reports no-longer-needed diagnostic suppression comments.

If you see a legitimate false positive error generated by pyright, please file a bug so we can address it.

**Comment by NeilGirdhar** [comment]
Awesome!   That completely satisfies my needs.  I will just do that.

**Comment by NeilGirdhar** [comment]
@erictraut Is there a way to have pyright warn on unused `$ pyright: ignore` directives?

**Comment by erictraut** [comment]
> This mechanism also works with the `reportUnnecessaryTypeIgnoreComment` option, which reports no-longer-needed diagnostic suppression comments.

**Comment by NeilGirdhar** [comment]
Sorry, missed that!

**Comment by NeilGirdhar** [comment]
Oh, but that triggers on all of the type ignores I added for mypy.  Is there a way to report only for unnecessary `# pyright: ignore` comments?

**Comment by erictraut** [comment]
No, `reportUnnecessaryTypeIgnoreComment` applies to `# type: ignore` and `# pyright: ignore`. 

If you want mypy-specific errors, you'll need to convince the mypy maintainers to implement that. The `# type: ignore` comment is defined in PEP 484 and isn't specific to any type checker.

**Comment by NeilGirdhar** [comment]
I know you're right.  You're an idealist like me :smile:
